sgkit_vcf/vcf_reader.py

```diff
@@ -0,0 +1,300 @@
+            variant_id_mask = variant_id == "."
+            variant_alleles = np.array(variant_alleles, dtype="O")
+
+            ds: xr.Dataset = create_genotype_call_dataset(
+                variant_contig_names=variant_contig_names,
+                variant_contig=variant_contig,
+                variant_position=variant_position,
+                variant_alleles=variant_alleles,
+                sample_id=sample_id,
+                call_genotype=call_genotype,
+                call_genotype_phased=call_genotype_phased,
+                variant_id=variant_id,
+            )
+            ds["variant_id_mask"] = (
+                [DIM_VARIANT],
+                variant_id_mask,
+            )
+            ds.attrs["max_variant_id_length"] = max_variant_id_length
+            ds.attrs["max_variant_allele_length"] = max_variant_allele_length
+
+            if first_variants_chunk:
+                # Enforce uniform chunks in the variants dimension
+                # Also chunk in the samples direction
+                encoding = dict(
+                    call_genotype=dict(chunks=(chunk_length, chunk_width, n_ploidy)),
+                    call_genotype_mask=dict(
+                        chunks=(chunk_length, chunk_width, n_ploidy)
+                    ),
+                    call_genotype_phased=dict(chunks=(chunk_length, chunk_width)),
+                    variant_allele=dict(chunks=(chunk_length, n_allele)),
+                    variant_contig=dict(chunks=(chunk_length,)),
+                    variant_id=dict(chunks=(chunk_length,)),
+                    variant_id_mask=dict(chunks=(chunk_length,)),
+                    variant_position=dict(chunks=(chunk_length,)),
+                    sample_id=dict(chunks=(chunk_width,)),
+                )
+
+                ds.to_zarr(output, mode="w", encoding=encoding)
+                first_variants_chunk = False
+            else:
+                # Append along the variants dimension
+                ds.to_zarr(output, append_dim=DIM_VARIANT)
+
+
+def vcf_to_zarr_parallel(
+    input: Union[PathType, Sequence[PathType]],
+    output: PathType,
+    regions: Union[None, Sequence[str], Sequence[Optional[Sequence[str]]]],
+    chunk_length: int = 10_000,
+    chunk_width: int = 1_000,
+    tempdir: Optional[Path] = None,
+) -> None:
+    """Convert specified regions of one or more VCF files to zarr files, then concat, rechunk, write to zarr"""
+
+    paths = vcf_to_zarrs(input, output, regions, chunk_length, chunk_width, tempdir)
+
+    ds = zarrs_to_dataset(paths, chunk_length, chunk_width)
+
+    # Ensure Dask task graph is efficient, see https://github.com/dask/dask/issues/5105
+    with dask.config.set({"optimization.fuse.ave-width": 50}):
+        ds.to_zarr(output, mode="w")
+
+    # Delete intermediate files from temporary directory
+    for path in paths:
+        shutil.rmtree(path)
+
+
+def vcf_to_zarrs(
+    input: Union[PathType, Sequence[PathType]],
+    output: PathType,
```

Comment:
Oh wait, this is never used?

Reply:
Good catch. Fixed in https://github.com/pystatgen/sgkit-vcf/pull/1/commits/926060bed74129481791c071edcb998f9cd5dfab